content/blog/aws-gke-token-exchange/index.md

```diff
@@ -0,0 +1,127 @@
+
+```bash
+$ gcloud services enable sts.googleapis.com
+$ gcloud services enable iamcredentials.googleapis.com
+$ gcloud resource-manager org-policies allow constraints/iam.workloadIdentityPoolAwsAccounts \
+    <aws-account-id> --organization=<gcp org>
+```
+
+Note: you will also need to ensure that you have the Workload Identity Pool Admin (roles/iam.workloadIdentityPoolAdmin) and Service Account Admin (roles/iam.serviceAccountAdmin) roles on the project.
+
+## Creating a workload identity provider
+
+Here we will create a workload identity provider for our token exchange with AWS
+
+First let's create the pool
+
+```bash
+$ gcloud iam workload-identity-pools create aws-pool \
+    --location="global" \
+    --description="Workload identity pool for aws connectivity." \
+    --display-name="AWS pool"
+```
+
+Then the provider
+
+```bash
+$ gcloud iam workload-identity-pools providers create-aws aws-test-account \
+    --location="global"  \
+    --workload-identity-pool="aws-pool" \
+    --account-id="<your account>" \
+    --display-name="Test AWS provider"  \
+    --description="The Identity Provider for AWS test service"
+```
+
+Note: if you would like to be more explicit about what aws role can access this workload identity add the following (replacing the account name and role)
+
+```bash
+--attribute-condition="'arn:aws:sts::000000000000:assumed-role/some-role' == attribute.aws_role"  \
+```
+
+## Combining workload federation and service accounts
+
+Now that we have the ability to gain an access token using the workload federation, we now need to allow the workload provider to assume the service account role.
+
+```bash
+$ gcloud iam service-accounts add-iam-policy-binding aws-service-account-demo@example-project.iam.gserviceaccount.com \
+   --role roles/iam.workloadIdentityUser \
+   --member "principalSet://iam.googleapis.com/projects/$gcp_project_number/locations/global/workloadIdentityPools/aws-pool/subject/arn:aws:sts::${aws_account_id}:assumed-role/$role_name"
+```
+
+Note: if you want to allow all resources in the workload identity pool to assume to the service account replace the --member field with the following
+
+```bash
+--member "principalSet://iam.googleapis.com/projects/$gcp_project_number/locations/global/workloadIdentityPools/aws-pool/*"
+```
+
+# Token exchange flow
+
+Combining all the steps configured and knowledge above, we are now ready to initiate the token exchange flow. Below is a flow chart of each of the steps.
+
+![Flow diagram](./flow.png)
+
+1. The users logs in to their aws ecosystem and receives a set of aws specific temporary credentials
+2. User signs their GetCallerIdenity request with their personal temporary credentials
+3. User sends a POST request to GCP; [https://sts.googleapis.com/v1/token](https://sts.googleapis.com/v1/token). This contains all the custom signed credentials in the requested payload. This will be used to verify the user's identity.
```

Comment:
This URL looks clickable, but isn't really. IMO it would make more sense to wrap it in backticks.

Reply:
Will do thank you